src/game.c

```diff
@@ -477,6 +477,132 @@ int game_do_command_p(int command, int *eax, int *ebx, int *ecx, int *edx, int *
+			// Get ride name
+			int ride_index = *edx & 0xFF;
+			rct_ride* ride = get_ride(ride_index);
+			char ride_name[128];
+			format_string(ride_name, ride->name, &ride->name_arguments);
+
+			char* args[2] = {
+				(char *) player_name,
+				ride_name
+			};
+
+			switch (command) {
+				case GAME_COMMAND_SET_RIDE_APPEARANCE: format_string(log_msg, STR_LOG_RIDE_APPEARANCE, args); break;
+				case GAME_COMMAND_SET_RIDE_VEHICLES: format_string(log_msg, STR_LOG_RIDE_VEHICLES, args); break;
+				case GAME_COMMAND_SET_RIDE_SETTING: format_string(log_msg, STR_LOG_RIDE_SETTINGS, args); break;
+			}
+
+			network_append_server_log(log_msg);
+		} else if (command == GAME_COMMAND_SET_RIDE_STATUS) {
+			// Get ride name
+			int ride_index = *edx & 0xFF;
+			rct_ride* ride = get_ride(ride_index);
+			char ride_name[128];
+			format_string(ride_name, ride->name, &ride->name_arguments);
+
+			char* args[2] = {
+				(char *) player_name,
+				ride_name
+			};
+
+			int status = *edx >> 8;
+			switch (status) {
+				case 0: format_string(log_msg, STR_LOG_RIDE_STATUS_CLOSED, args); break;
+				case 1: format_string(log_msg, STR_LOG_RIDE_STATUS_OPEN, args); break;
+				case 2: format_string(log_msg, STR_LOG_RIDE_STATUS_TESTING, args); break;
+			}
+
+			network_append_server_log(log_msg);
+		} else if (command == GAME_COMMAND_SET_RIDE_PRICE) {
+			// Get ride name
+			int ride_index = *edx & 0xFF;
+			rct_ride* ride = get_ride(ride_index);
+			char ride_name[128];
+			format_string(ride_name, ride->name, &ride->name_arguments);
+
+			// Format price
+			int price_args[1] = {*edi};
+			char price_str[16];
+			format_string(price_str, STR_BOTTOM_TOOLBAR_CASH, price_args);
+
+			// Log change in primary or secondary price
+			char log_msg[256];
+			char* args[3] = {
+				(char *) player_name,
+				ride_name,
+				price_str
+			};
+
+			if (*edx >> 8 == 0) {
+				format_string(log_msg, STR_LOG_RIDE_PRICE, args);
+			} else if (*edx >> 8 == 1) {
+				format_string(log_msg, STR_LOG_RIDE_SECONDARY_PRICE, args);
+			}
+			
+			network_append_server_log(log_msg);
+		} else if (command == GAME_COMMAND_SET_PARK_OPEN) {
+			// Log change in park open/close
+			char log_msg[256];
+			char* args[1] = {
+				(char *) player_name
+			};
+			if (*edx >> 8 == 0) {
+				format_string(log_msg, STR_LOG_PARK_OPEN, args);
+			} else if (*edx >> 8 == 1) {
+				format_string(log_msg, STR_LOG_PARK_CLOSED, args);
+			}
+
+			network_append_server_log(log_msg);
+		} else if (command == GAME_COMMAND_SET_PARK_ENTRANCE_FEE) {
+			// Format price
+			int price_args[1] = {*edi};
+			char price_str[16];
+			format_string(price_str, STR_BOTTOM_TOOLBAR_CASH, price_args);
+
+			// Log change in park entrance fee
+			char log_msg[256];
+			char* args[2] = {
+				(char *) player_name,
+				price_str
+			};
+
+			format_string(log_msg, STR_LOG_PARK_ENTRANCE_FEE, args);
+			network_append_server_log(log_msg);
+		}
+	}
```

Comment:
Please move this to its own function, otherwise game_command just becomes far too big.